Day 02

jamt23 opened the pull request:
## Overview

I attempted to copy the site.png image in my own page. I was able to do it in less than 40 lines of code. I had troubles finding what font you used, however, this is the closest I could find. The boldness of the font I used was a bit thicker. Also, during testing, I was not able to get the page to show on github pages. The readme is what shows even though I put "./index.html". 

Please advise on what is wrong. On VS code it worked for me.
<img width="1440" alt="Screen Shot 2022-12-24 at 5 18 08 PM" src="https://user-images.githubusercontent.com/120585837/209453111-b74b5335-22d8-428e-b068-963a6a935cd3.png">

jamt23 committed:
This is my day 02 challenge first upload attempt

jamt23 committed:
added link to readme

nilsonmolina commented:
Nice job!

The link does not work, because this code is currently just a PR, github has not published your `index.html` yet. We would need to merge this PR and then all that code gets published to Github Pages. So this is working as intended!

You also got the font right, great job! I'd say the only things that need to be fixed here are the edges of the "Welcome to Day 02" block. The are different between the two. Also, to a minor degree, the left and right padding of that same block is a little off too.

jamt23 committed:
added border radius and also adjusted the padding

jamt23 commented:
I did not notice the border at all. I have updated the code to add the border. Also, the padding was seen but not understood as it had the same numbers but would not show the same. I have changed the numbers so it looks more even instead of reading more even in the code.

nilsonmolina commented on challenges/day-02/index.html:
```diff
@@ -0,0 +1,29 @@
+    <style>
+        h1 {
+            width: 20%;
+            height: 20%;
```
Using a value like 20% here could cause issues when resizing the window. Give it a try and you'll see that it breaks the title up into multiple lines. Sometimes thats exactly what you want, but in this case, maybe not the best solution. I think a better value would be `fit-content`. Give it a try and let me know what you think.

jamt23 replied:
Ah I see how that works. I noticed when resizing. Thank you for teaching me that trick as every device has a different aspect ratio and not always in full screen.

nilsonmolina commented on challenges/day-02/index.html:
```diff
@@ -0,0 +1,29 @@
+            width: 20%;
+            height: 20%;
+            margin: 20px;
+            padding: 5px 5px 5px 15px;
```
When applying margin/padding, you generally want to have similar values for left/right. In this case, I would use the following values: `5px 15px`. That way the top/bottom are 5px and the left/right are both 15px. Keeps things look a lot more consistent. 

Looking over my actual code, I ended up using `padding: 15px 20px`... take a look at both and see which one you prefer.

jamt23 replied:
I attempted both sizes and went with the one you did just to keep the same code. I tried both and the 5 x 15 is too small. It depends on everything else as well I guess.

nilsonmolina commented on challenges/day-02/index.html:
```diff
@@ -0,0 +1,29 @@
+            padding: 5px 5px 5px 15px;
+            background-color: rgba(255, 255, 255, 0.2);
+            border-color: white;
+            color: white;
```
This is a super minor nitpick, and more informational than anything else. Color theory is a huge topic and something that I don't think I'll ever fully understand, but one concept/idea I come across often is to avoid using pure white or black in web design. Has something to do with contrast and helps reduce the harshness that pure white can sometimes cause on our eyes. It's always better to use a shade instead. This is not a hard requirement, and using white here is perfectly acceptable.

In my case I used the following value: `#ddd`. That ends up being a shade of white'ish grey. The above value is a hexadecimal value often used for colors. There is a recent trend to move to HSL instead of hex, but I recommend learning hex first as it is still the standard pretty much everywhere. 

Hexadecimal goes from 0-F (0, 1, 2, 3, 4, 5, 6, 7, 8, 9, A, B, C, D, E, F). The way hex colors work is you assign a value for each of the RGB (Red, Green, Blue) values. So if you want a perfect red, that would be `#F00`. If you want Purple, you would do `#F0F`, since purple is just red and blue combined. RGB is additive so if you want white, you essentially turn on all colors `#FFF`, and for black you turn off all colors `#000`. Lastly, sometimes instead of 3 digits, you may see 6 digits (ie #F6C509). That just gives you more granularity in your color choices but it essentially works the same (#RRGGBB), so #FFFFFF is equivalent to #FFF. 

Hope that helps, we can chat more about it if you have questions, just wanted to give a super quick rundown of how colors work.

jamt23 replied:
Ok I tried it out and I see. I was learning about this in one of the courses however, it did not explain it as well. How would I go about doing that and then adding the transparency to it though? I was confused about that part and had to google it when I first did it and that was the way I found. I will try to find out first but if you can run that by me when we next chat it would be super helpful.

nilsonmolina replied:
I was only referring to the `color` property here. Border color is not necessary and the background was perfectly fine since you were using an opacity (or alpha) value already.

And yeah, we can totally chat about this in more detail if you'd like.

jamt23 replied:
Ah ok. I went ahead and got rid of the border-color and it still looks the same. Got it. I would like to chat more about it so I can understand it better. Also, I did try to change the format for the background-color so that it is in the same hexadecimal format.

jamt23 committed:
changed width and height value to 'fit-content'

jamt23 committed:
adjusted the padding and the color codes

jamt23 committed:
removed border-color
